### PR TITLE
Enhance integration tests

### PR DESCRIPTION
**Pull request by keradus**
Any help appreciated!

### CODE REVIEW

File: Symfony/CS/Fixer/Contrib/StrictFixer.php [hunk 1]
```diff
@@ -55,13 +55,4 @@ public function getDescription()
...
         return 'Comparison should be strict. Warning! This could change code behavior.';
     }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getPriority()
-    {
-        // should be run after the StandardizeNotEqualFixer
-        return -10;
```
Comment (keradus) [inline Symfony/CS/Fixer/Contrib/StrictFixer.php]: actually no need for this, as `<>` and `!=` are _THE SAME_ token kind for tokenizer :/

**Comment by keradus** [comment]
Odd.. it passes locally...

File: Symfony/CS/Tests/AbstractIntegrationTest.php [hunk 2]
```diff
@@ -175,7 +175,20 @@ protected function doTestIntegration($testFileName, $testTitle, $fixers, array $
         }
 
         $this->assertNotEmpty($changed, sprintf('Expected changes made to test "%s" in "%s".', $testTitle, $testFileName));
-        $this->assertSame($expected, file_get_contents($tmpFile), sprintf('Expected changes do not match result for "%s" in "%s".', $testTitle, $testFileName));
+        $fixedInputCode = file_get_contents($tmpFile);
+        $this->assertSame($expected, $fixedInputCode, sprintf('Expected changes do not match result for "%s" in "%s".', $testTitle, $testFileName));
+
+        $priorities = array_map(function (FixerInterface $fixer) { return $fixer->getPriority(); }, $fixers);
```
Comment (SpacePossum) [inline Symfony/CS/Tests/AbstractIntegrationTest.php]: are one line closure allowed for PSR2?

File: Symfony/CS/Tests/AbstractIntegrationTest.php [hunk 3]
```diff
@@ -175,7 +175,20 @@ protected function doTestIntegration($testFileName, $testTitle, $fixers, array $
...
+        $this->assertNotCount(1, array_unique($priorities), 'All used fixers must not have the same priority, integration tests should cover fixers with different priorities.');
```
Comment (SpacePossum) [inline Symfony/CS/Tests/AbstractIntegrationTest.php]: is 0 ok (i.e. shouldn't be greater than 1)?

File: Symfony/CS/Tests/AbstractIntegrationTest.php (hunk 2 above)
Comment (gharlan) [inline Symfony/CS/Tests/AbstractIntegrationTest.php]: no

**Comment by SpacePossum** [comment]
Like the idea, changes look good :+1: 
you call for help is for finding why the test fails @ travis or is there more work to be done?

File: Symfony/CS/Tests/AbstractIntegrationTest.php (hunk 2 above)
Comment (GrahamCampbell) [inline Symfony/CS/Tests/AbstractIntegrationTest.php]: we currently don't fix them though

File: Symfony/CS/Tests/AbstractIntegrationTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/AbstractIntegrationTest.php]: it cannot be 0, it would means your `$fixers` array is empty

**Comment by keradus** [comment]
Firstly it means I asked for help covering the rest of priority relations, now it means Travis too (Travis is red... AppVeyor is green... :|) ;)

**Comment by keradus** [comment]
I have removed non-passing test to allow merge the enhanced test method to be merged, and then we will work on that integration test case again.

**Comment by SpacePossum** [comment]
>I have removed non-passing test

Travis and HHVM don't think so :(

**Comment by keradus** [comment]
Done.
This is why it is good to have atomic integration-tests.
Bugfix backported to 1.11.x: #1595

**Pull request merged**